python/ray/utils.py

```diff
@@ -489,6 +490,15 @@ def get_system_memory():
     return psutil_memory_in_bytes
 
 
+def get_num_cpus():
+    try:
```

Comment:
also, does https://stackoverflow.com/a/55423170/2213289 sched_affinity work too? would that be more pythonic?

Reply:
haven't tested, but docker usually requires some wrangling to get affinity to work. (also the `os` module and docker just generally don't play nice).